assets/js/modules/analytics-4/datastore/audiences.js

```diff
@@ -203,93 +205,114 @@ const baseActions = {
+				// Audience slugs to sort by:
+				const sortedSlugs = [ 'new-visitors', 'returning-visitors' ];
 
-			const sortedSiteKitAudiences = siteKitAudiences.sort(
-				( audienceA, audienceB ) => {
-					const indexA = sortedSlugs.indexOf(
-						audienceA.audienceSlug
-					);
-					const indexB = sortedSlugs.indexOf(
-						audienceB.audienceSlug
-					);
+				const sortedSiteKitAudiences = siteKitAudiences.sort(
+					( audienceA, audienceB ) => {
+						const indexA = sortedSlugs.indexOf(
+							audienceA.audienceSlug
+						);
+						const indexB = sortedSlugs.indexOf(
+							audienceB.audienceSlug
+						);
 
-					return indexA - indexB;
-				}
-			);
+						return indexA - indexB;
+					}
+				);
 
-			const audienceResourceNames = sortedSiteKitAudiences
-				.slice( 0, MAX_INITIAL_AUDIENCES - configuredAudiences.length )
-				.map( ( { name } ) => name );
+				const audienceResourceNames = sortedSiteKitAudiences
+					.slice(
+						0,
+						MAX_INITIAL_AUDIENCES - configuredAudiences.length
+					)
+					.map( ( { name } ) => name );
 
-			configuredAudiences.push( ...audienceResourceNames );
+				configuredAudiences.push( ...audienceResourceNames );
+			}
 		}
 
 		if ( configuredAudiences.length === 0 ) {
+			const audiencesToCreate = failedSiteKitAudienceResourceNames?.length
+				? failedSiteKitAudienceResourceNames
+				: [ 'new-visitors', 'returning-visitors' ];
+
 			// If there are no configured audiences by this point, create the "new-visitors" and "returning-visitors" audiences,
 			// and add them to the configured audiences.
 			const [ newVisitorsResult, returningVisitorsResult ] =
 				yield commonActions.await(
-					Promise.all( [
-						dispatch( MODULES_ANALYTICS_4 ).createAudience(
-							SITE_KIT_AUDIENCE_DEFINITIONS[ 'new-visitors' ]
-						),
-						dispatch( MODULES_ANALYTICS_4 ).createAudience(
-							SITE_KIT_AUDIENCE_DEFINITIONS[
-								'returning-visitors'
-							]
-						),
-					] )
+					Promise.all(
+						audiencesToCreate.map( ( audienceSlug ) => {
+							return dispatch(
+								MODULES_ANALYTICS_4
+							).createAudience(
+								SITE_KIT_AUDIENCE_DEFINITIONS[ audienceSlug ]
+							);
+						} )
+					)
 				);
 
-			// TODO: Full error handling will be implemented via https://github.com/google/site-kit-wp/issues/8134.
-			if ( ! newVisitorsResult.error ) {
+			const failedAudiencesToRetry = [];
+
+			if ( newVisitorsResult.error ) {
+				failedAudiencesToRetry.push( 'new-visitors' );
+			} else {
 				configuredAudiences.push( newVisitorsResult.response.name );
 			}
 
-			if ( ! returningVisitorsResult.error ) {
+			if ( returningVisitorsResult.error ) {
+				failedAudiencesToRetry.push( 'returning-visitors' );
+			} else {
 				configuredAudiences.push(
 					returningVisitorsResult.response.name
 				);
 			}
 
+			if ( failedAudiencesToRetry.length > 0 ) {
+				return {
+					failedSiteKitAudienceResourceNames: failedAudiencesToRetry,
```

Comment:
As above this name is incorrect:

```suggestion
					failedSiteKitAudienceSlugs: failedAudiencesToRetry,
```

Reply:
Updated.